tests/live_tests.py

```diff
@@ -0,0 +1,119 @@
+if not args.tag_name: args.tag_name = '"fake_tag_name"~'
+if not args.key: args.key = 'fake_key"%'
+if not args.value: args.value = 'fake_value'
+
+FAKE_METRIC_NAME, FAKE_TAG, FAKE_KEY, FAKE_VALUE = \
+    args.metric_name, args.tag_name, args.key, args.value
+
+CLIENT_NAME = 'sfx'
+TOKEN = args.token
+gT, sT, dT, uT = 'get_tag', 'search_tags', 'delete_tag', 'update_tag'
+sD, uD, gD = 'search_dimensions', 'update_dimension', 'get_dimension'
+
+# import the correct signalfx
+this_path = os.path.dirname(os.path.realpath(__file__))
+desired_path = this_path.replace('tests', '')
+if desired_path not in sys.path: sys.path.insert(0, desired_path)
+import signalfx
+exec(CLIENT_NAME + '=signalfx.SignalFx(TOKEN)')
+
+def send_data_example(client_name, num_pts=10):
+    sess = eval(client_name)
+    init_stamp = 1453960607000
+    init_value = 500
+    count = 0
+    while count < num_pts:
+        i, j, k = random.randint(-20, 20), random.randint(1,500), random.randint(1,3)
+        sess.send(
+            gauges=[
+                {
+                    'metric': FAKE_METRIC_NAME,
+                    'value' : init_value + i,
+                    'timestamp': init_stamp + j,
+                    'dimensions': {'host': 'server' + str(k), 'host_ip': '1.2.3.4',
+                                   FAKE_KEY : FAKE_VALUE}
+                }
+            ])
+        atexit.register(sess.stop)
+        count += 1
+    print 'data sent, waiting for update...'
+    time.sleep(15)
+
+def test_func(client_name, func_name, sleep_time=5, msg='is being tested!',
+              func_args=None, **kwargs):
+    return_value = None
+    if func_name in dir(eval(client_name)):
+        try:
+            func = eval(client_name + '.' + func_name)
+            return_value = func(*func_args, **kwargs)
+            print func_name, msg, '\n', return_value, '\n'
+            time.sleep(sleep_time)
+        except requests.exceptions.HTTPError, e:
+            print func_name, e.message
+    else:
+        print func_name, 'not available'
+    return return_value
+
+def apply_sequence(client_name, sequence, f_args, seq_msg=None, **kwargs):
+    print sequence, 'expected behavior is', seq_msg
+    for item in sequence:
+        test_func(client_name, item, func_args=f_args, **kwargs)
+
+def main():
+    send_data_example(CLIENT_NAME)
+    time.sleep(20)
+    apply_sequence(CLIENT_NAME, [gT, sT, dT, uT, gT, sT, dT, gT, sT, uT,
+                                 gT, sT, dT], (FAKE_TAG,), timeout=15)
+    test_func(CLIENT_NAME, sD, func_args=[FAKE_KEY], timeout=15)
+    apply_sequence(CLIENT_NAME, [uD, gD], (FAKE_KEY, FAKE_VALUE), timeout=15)
+    test_func(CLIENT_NAME, sD, func_args=[FAKE_VALUE], timeout=15)
+    test_func(CLIENT_NAME, 'get_time_series_window',
+              func_args=['sf_source:analytics-gauss AND sf_metric:"jvm.heap.used"',
+                         450000000000, 1455320231110], resolution=3600000, timeout=20)
```

Comment:
In future, searching for this time range can be heavy. how about time.time()*1000 - (few seconds)?